syncer/syncer.go

```diff
@@ -2330,88 +2369,98 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext, o
 			if _, ok := stmt.(ast.DDLNode); ok {
-				tables, err3 := parserpkg.FetchDDLTables(usedSchema, stmt, s.SourceTableNamesFlavor)
+				tables, err3 := parserpkg.FetchDDLTables(qec.ddlSchema, stmt, s.SourceTableNamesFlavor)
 				if err3 != nil {
 					continue
 				}
 				// nolint:errcheck
-				s.onlineDDL.Apply(ec.tctx, tables, "", stmt)
+				s.onlineDDL.Apply(qec.tctx, tables, "", stmt)
 			}
 		}
-		return s.recordSkipSQLsLocation(&ec)
+		return s.recordSkipSQLsLocation(qec.eventContext)
 	}
 	if !parseResult.isDDL {
 		// skipped sql maybe not a DDL
 		return nil
 	}
 
-	if ec.shardingReSync != nil {
-		ec.shardingReSync.currLocation = *ec.currentLocation
-		if binlog.CompareLocation(ec.shardingReSync.currLocation, ec.shardingReSync.latestLocation, s.cfg.EnableGTID) >= 0 {
-			ec.tctx.L().Info("re-replicate shard group was completed", zap.String("event", "query"), zap.String("statement", originSQL), zap.Stringer("re-shard", ec.shardingReSync))
-			err2 := ec.closeShardingResync()
+	if qec.shardingReSync != nil {
+		qec.shardingReSync.currLocation = *qec.currentLocation
+		if binlog.CompareLocation(qec.shardingReSync.currLocation, qec.shardingReSync.latestLocation, s.cfg.EnableGTID) >= 0 {
+			qec.tctx.L().Info("re-replicate shard group was completed", zap.String("event", "query"), zap.Stringer("queryEventContext", qec))
+			err2 := qec.closeShardingResync()
 			if err2 != nil {
 				return err2
 			}
 		} else {
 			// in re-syncing, we can simply skip all DDLs,
 			// as they have been added to sharding DDL sequence
 			// only update lastPos when the query is a real DDL
-			*ec.lastLocation = ec.shardingReSync.currLocation
-			ec.tctx.L().Debug("skip event in re-replicating sharding group", zap.String("event", "query"), zap.String("statement", originSQL), zap.Reflect("re-shard", ec.shardingReSync))
+			*qec.lastLocation = qec.shardingReSync.currLocation
+			qec.tctx.L().Debug("skip event in re-replicating sharding group", zap.String("event", "query"), zap.Stringer("queryEventContext", qec))
 		}
 		return nil
 	}
 
-	ec.tctx.L().Info("", zap.String("event", "query"), zap.String("statement", originSQL), zap.String("schema", usedSchema), zap.Stringer("last location", ec.lastLocation), log.WrapStringerField("location", ec.currentLocation))
-	*ec.lastLocation = *ec.currentLocation // update lastLocation, because we have checked `isDDL`
+	qec.tctx.L().Info("", zap.String("event", "query"), zap.Stringer("queryEventContext", qec))
+	*qec.lastLocation = *qec.currentLocation // update lastLocation, because we have checked `isDDL`
 
 	// TiDB can't handle multi schema change DDL, so we split it here.
 	// for DDL, we don't apply operator until we try to execute it. so can handle sharding cases
 	// We use default parser because inside function where need parser, sqls are came from parserpkg.SplitDDL, which is StringSingleQuotes, KeyWordUppercase and NameBackQuotes
 	// TODO: save stmt, tableName to avoid parse the sql to get them again
```

Comment:
can we resolve some part of this TODO?

Reply:
We don't resolve this TODO this pr. Maybe will resolve it in https://github.com/pingcap/dm/pull/2090.